rp2040-hal/src/clocks/mod.rs

```diff
@@ -301,6 +335,63 @@ impl ClocksManager {
             .configure_clock(&self.system_clock, self.system_clock.freq())
     }
 
+    /// Measure the frequency of the given clock source by approximation.
+    pub fn measure_frequency<C: ClockSource>(
+        &mut self,
+        _trg_clk: C,
+        accuracy: FCAccuracy,
+    ) -> Result<HertzU32, ClockDiedError> {
```

Comment:
I had some problems calling this function at all:
```
error[E0505]: cannot move out of `clocks.adc_clock` because it is borrowed
  --> src/main.rs:58:39
   |
35 |     let mut clocks = init_clocks_and_plls(
   |         ---------- binding `clocks` declared here
...
58 |     let m1 = clocks.measure_frequency(clocks.adc_clock, FCAccuracy::_1kHz );
   |              ------ ----------------- ^^^^^^^^^^^^^^^^ move out of `clocks.adc_clock` occurs here
   |              |      |
   |              |      borrow later used by call
   |              borrow of `clocks` occurs here
```

How is it meant to be used?
(I worked around that by changing the signature:
```
    pub fn measure_frequency<C: ClockSource>(
        &self,
        _trg_clk: &C,
        accuracy: FCAccuracy,
    ) -> Result<HertzU32, ClockDiedError> {
```
Which works, but I guess the `&mut self` is desirable to guarantee exclusive access.

Reply:
Yeh, let me remove that part of the PR then. It's causing more troubles than anything.
I thought that'd be an incremental step on the clock update (rather than the monolithic thing that GPIO were and PIO will be :( )

Reply:
Don't get me wrong, I like the feature. But it would be nice to be able to call the function :-)

Keeping it separate from fixing the VCO frequency is a good idea, though. The minimal fix, updating the settings for the USB PLL, should be in its own PR.

Reply:
This could work:
```
     pub fn measure_frequency(                                  
         &mut self,                   
         clock: FC0_SRC_A,                                                 
         accuracy: FCAccuracy, 
     ) -> HertzU32 {                                                                                
```
Little bit ugly at the call site, but no ownership issues:
```
let m1 = clocks.measure_frequency(AdcClock::FCOUNTER_SRC, FCAccuracy::_1kHz );
```